modules/commsInterface/commsInterface.py

```diff
@@ -0,0 +1,74 @@
+import usb.core
+import usb.util
+import serial
+
+class CommsInterface:
+    def __init__(self, type: bool):
+        """
+        Initializes parameters for USB or UART
+
+        Parameters
+        ----------
+        type: bool
+            Boolean used to specify selection of USB or UART;
+            0|False for USB, 1|True for UART
+        """
+        self.uart_or_usb = type
+    
+    def create_end_point_FC(self):
+        """
+        Issues a read command in either serial (UART) or pyusb (USB).
+        EndpointId can be the specific serial port to use or the usb endpoint to be used. 
+        """
+        if self.uart_or_usb == 0:
+            if self.idVendor and self.idProduct is not None:
+                self.dev = usb.core.find(idVendor=self.idVendor, idProduct=self.idProduct)
+            else:
+                self.dev = usb.core.find()
+            if self.dev is None:
+                raise ValueError('Device Not Found')
+
+            self.ep = self.dev[0].interfaces()[0].endpoints()[0]
+            self.dev.set_configuration()
+
+            return self.ep.bEndpointAddress
+        elif self.uart_or_usb == 1:
+            return serial.Serial(self.uart_port, self.baudrate)
+
+    def read(self, endpointId):
+        """
+        Issues a write command in either serial or pyusb.
+        Data parameter should be a buffer. 
+        """
+        if self.uart_or_usb == 0:
+            return self.dev.read(endpointId, self.ep.wMaxPacketSize)
+        elif self.uart_or_usb == 1:
+            read_data = endpointId.readline()
+            return str(read_data[0:len(read_data)].decode('utf-8'))
```

Comment:
I don't think you should decode this data as a string. Reason being, Firmware and CV will come up with an API that is likely a raw byte stream (i.e. the number '232' is going to be '0xE8', which will decode to some random ASCII character).

Reply:
Good catch, but then we need to develop a lookup for all our mappings from raw bytes to values?

Reply:
I don't understand what you mean by lookup? The way you'd typically do this is have a python struct definition on the computer side detailing what to expect in byte form. Something like
- 4x bytes unsigned integers, 1x float big endian etc
that is developed as a result of the FW/CV API. Then, you'd do struct.unpack() on the bytes object and store the results in various variables.

I guess this means there will be a master lookup somewhere, this is typically done with python structs though https://docs.python.org/3/library/struct.html